### PR TITLE
Allow dockerfile in hat path

### LINKED ISSUE #189: hats: support dockerfile paths
I was writing the documentation updates and realized that it is kind of weird that it requires a directory instead of accepting both. Is this an improvement that should be made? If so, I can create a PR for it.

### LINKED ISSUE #193: Why are go modules turned off?
in the .sail/Dockerfile, go modules are turned off, but there is a go.mod file and go.sum file.

[Go Github](https://github.com/google/go-github) not install because of it.

### PR DESCRIPTION
**Pull request by roberthmiller**
Resolves #189

### CODE REVIEW

**Comment by roberthmiller** [comment]
Do you guys think that the `chown` command should be run in the `codercom/ubuntu-dev-go container`?

File: .sail/Dockerfile [hunk 1]
```diff
@@ -8,12 +8,12 @@ RUN curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.34.0/install.sh | b
 
 LABEL project_root "~/go/src/go.coder.com"
 
-# Modules break much of Go's tooling.
-ENV GO111MODULE=off
+ENV GO111MODULE=on
```
Comment (deansheather) [inline .sail/Dockerfile]: I think this should be submitted in a separate PR for discussion as this is a controversial change. @ammario might have something to say about it, since he was the one who initially added this according to git blame: https://github.com/cdr/sail/pull/49/files#diff-e963b843e440ee24f53402574e0e3dd6.

File: .sail/Dockerfile [hunk 2]
```diff
@@ -8,12 +8,12 @@ RUN curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.34.0/install.sh | b
...
 
 # Install the latest version of Hugo.
 RUN wget -O /tmp/hugo.deb https://github.com/gohugoio/hugo/releases/download/v0.55.4/hugo_extended_0.55.4_Linux-64bit.deb && \
   sudo dpkg -i /tmp/hugo.deb && \
   rm -f /tmp/hugo.deb
 
+RUN mkdir -p ~/go && sudo chown -R user:root ~/go
```
Comment (deansheather) [inline .sail/Dockerfile]: This should also be submitted in a separate PR for discussion. Also, if it affects most/all Go repos using Sail then this could be added to the ubuntu-dev-go image directly.

**Comment by deansheather** [comment]
Also, please sign your commits with PGP so we can add them to sail. :key:

**Comment by roberthmiller** [comment]
@deansheather I think this PR is good to go

File: .sail/Dockerfile [hunk 3]
```diff
@@ -8,7 +8,6 @@ RUN curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.34.0/install.sh | b
 
 LABEL project_root "~/go/src/go.coder.com"
 
-# Modules break much of Go's tooling.
```
Comment (nathanpotter) [inline .sail/Dockerfile]: We should add this comment back before merging

Comment (roberthmiller) [reply .sail/Dockerfile]: Completely forgot about that... Will do it right now

**Comment by roberthmiller** [comment]
@nathanpotter @deansheather This PR is ready to be merged
